flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/PreValidateReWriter.scala

```diff
@@ -45,10 +46,13 @@ class PreValidateReWriter(
     val typeFactory: RelDataTypeFactory) extends SqlBasicVisitor[Unit] {
   override def visit(call: SqlCall): Unit = {
     call match {
-      case r: RichSqlInsert if r.getStaticPartitions.nonEmpty
-        && r.getSource.isInstanceOf[SqlSelect] =>
-        appendPartitionProjects(r, catalogReader, typeFactory,
-          r.getSource.asInstanceOf[SqlSelect], r.getStaticPartitions)
+      case r: RichSqlInsert if r.getStaticPartitions.nonEmpty => r.getSource match {
+        case select: SqlSelect =>
+          appendPartitionProjects(r, catalogReader, typeFactory, select, r.getStaticPartitions)
+        case source =>
+          throw new ValidationException(
+            s"Only support static partitions with select, current is: $source")
```

Comment:
```suggestion
            s"INSERT INTO <table> PARTITION statement only support SELECT clause for now, '$source' is not supported yet.")
```

What about improving the message to above?